dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/security/TwoFactorController.java

```diff
@@ -80,6 +82,32 @@
 
     private final SystemSettingManager systemSettingManager;
 
+    /**
+     * @deprecated Use {@link #generateQRCode}.
+     * @param currentUser
+     * @return
+     * @throws WebMessageException
+     */
+    @Deprecated( since = "2.39" )
+    @GetMapping( value = "/qr", produces = APPLICATION_JSON_VALUE )
+    @ResponseStatus( HttpStatus.ACCEPTED )
+    @ResponseBody
+    public Map<String, Object> getQrCode( @CurrentUser User currentUser )
+            throws WebMessageException
+    {
+        if ( currentUser == null )
+        {
+            throw new WebMessageException( conflict( ErrorCode.E3027.getMessage(), ErrorCode.E3027 ) );
+        }
+
+        String appName = systemSettingManager.getStringSetting( SettingKey.APPLICATION_TITLE );
```

Comment:
Have a look at the sonarcloud code smells; This line is not used and can be removed, and you can also resolve the smell with "deprecated" :)

Reply:
This method will be removed before release, when UI is migrated.